querybook/server/logic/datadoc.py

```diff
@@ -248,6 +248,59 @@ def clone_data_doc(id, owner_uid, commit=True, session=None):
     return new_data_doc
 
 
+@with_session
+def restore_data_doc_from_commit(
+    datadoc_id: int, commit_datadoc: DataDoc, commit=True, session=None
+) -> DataDoc:
+    data_doc = get_data_doc_by_id(datadoc_id, session=session)
+    assert data_doc is not None, "DataDoc not found"
+
+    # Update the DataDoc's title and meta
+    data_doc = update_data_doc(
+        datadoc_id,
+        title=commit_datadoc.title,
+        meta=commit_datadoc.meta,
+        commit=False,
+        session=session,
+    )
+
+    # Delete existing DataDocCells and DataCells
+    for existing_cell in data_doc.cells:
+        delete_data_doc_cell(
+            data_doc_id=data_doc.id,
+            data_cell_id=existing_cell.id,
+            commit=False,
+            session=session,
+        )
```

Comment:
instead of deleting all datacells and creating  them again, would it be better to keep those data cells if the cell id remains same, but only update the cell content?  sometimes the link people shares will include the cellId

Reply:
I think maybe keep current approach since updating existing cells might result in data integrity issues as restoring datadoc involve adding/deleting/updating cells and will be harder to maintain cell order